### ISSUE TITLE
SQL should be logged at debug level

### ISSUE THREAD

**Issue by rgamage** [aspnet/EntityFrameworkCore#14523]
Currently, if we implement logging filtered at the Information level, we get details of every EF SQL query.  This is clearly debug or trace level data, not Information.  It clutters everyone's logs, and is useless because the data is hiding the interesting things like id numbers, field values, etc.

What I'd like to see is these EF SQL details to be down-graded to Debug level.

**Comment by ajcvickers** [comment]
@DamianEdwards @davidfowl TL;DR Please let us know if we should not change the default log level for SQL to `debug`.

We've seen this feedback (SQL should be logged at debug) fairly consistently, and I think the general consensus on the team is we should change the level to debug. However, the decision to log SQL at info level was made by Project K at a higher level, so checking in with you guys before pulling the trigger.

**Comment by ajcvickers** [comment]
@rgamage The default experience when running an ASP.NET app in production is to log at Warning level. Logging at Info level is the default only for development environments.

With the implementation of #14970 it is now possible to easily configure any EF event to log at any level. This is making us revisit again whether changing the default level for SQL is the correct thing to do. Can you provide some more details of when and why you are using Info level logging?

/cc @divega @Eilon @DamianEdwards

**Comment by fschlaef** [comment]
For anyone wanting to re-enable query logging without embedding configuration in `DbContext` (with Serilog for example, which is what I'm doing), the override you're looking for is `Microsoft.EntityFrameworkCore.Database.Command`

Excerpt from my Serilog conf :

```json
"Serilog": {
    "MinimumLevel": {
      "Default": "Debug",
      "Override": {
        "System": "Information",
        "Microsoft": "Information",
        "Microsoft.EntityFrameworkCore.Database.Command": "Debug"
      }
    }
}
```

This pretty much restores the old behavior, while maintaining control over what is logged in config files.

**Comment by JonPSmith** [comment]
I agree with reducing the logging load because I have seen the impact of looging on performance. However I do think that having access to the database commands (typically SQL) that EF Core produces is very important. This is a great way to understand a database access, especially if you are looking to performance tune things. I recommend this to everyone who is concerned with EF Core performance.

The problems I see is the simple solution of setting the LogLevel to `Debug` would include a whole load of other Debug information (well, it does when I tried it). I am therefore looking at ways to isolate the database commands. Here are some solutions I came up with:

1. Use @fschlaef solution, i.e. filter by the EventId/Name. That works but needs a more complex setup.
2. My favourite would be all existing `Debug` level logging, apart from `Executing DbCommand` to the `Trace` level. I realise this might be too bigger change inside EF Core, but it would make life easier for EF Core users.
3. The final approach is to publicise the `DiagnosticSource` for getting database commands. I have no experience of using `DiagnosticSource` so I don't know if this is difficult for developers to use, but it was recommended to me in issue #15085.

**Comment by ajcvickers** [comment]
@JonPSmith As [documented](https://docs.microsoft.com/en-us/ef/core/what-is-new/ef-core-3.0/breaking-changes#query-execution-is-logged-at-debug-level), you can change the level of only this event to whatever you want with something like this:
```C#
protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
    => optionsBuilder
        .UseSqlServer(connectionString)
        .ConfigureWarnings(c => c.Log((RelationalEventId.CommandExecuting, LogLevel.Info)));
```

**Comment by JonPSmith** [comment]
@ajcvickers, Sorry, I read the headline and didn't read further - my bad. 

That is an acceptable fix and I'll try to publicise that when I'm writing about EF Core 3.

**Comment by ajcvickers** [comment]
Note: we are reverting this breaking change. See #15888

**Comment by ErikEJ** [comment]
Also update breaking changes docs?

**Comment by ajcvickers** [comment]
@ErikEJ Done, but haven't sent the PR yet.
